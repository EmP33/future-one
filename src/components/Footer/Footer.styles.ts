import styled from "styled-components";

export const Wrapper = styled.footer`
  width: 100%;
  background-color: var(--color-text);
  color: var(--color-white);
  padding: 4rem 0 1.5rem;

  @media only screen and (max-width: 600px) {
    padding: 4rem 0 0.5rem;
  }

  @media only screen and (max-width: 400px) {
    padding: 2.5rem 0 0.5rem;
  }

  .footer-wrapper {
    max-width: 85%;
    margin: 0 auto;

    @media only screen and (max-width: 1200px) {
      max-width: 90%;
    }

    @media only screen and (max-width: 900px) {
      max-width: 95%;
    }
  }

  .footer-content {
    display: grid;
    grid-template-columns: repeat(2, max-content) 1fr;
    column-gap: 7.5rem;
    justify-items: center;

    @media only screen and (max-width: 800px) {
      column-gap: 3rem;
    }

    @media only screen and (max-width: 600px) {
      grid-template-columns: 1fr 1fr;
    }

    @media only screen and (max-width: 400px) {
      grid-template-columns: 1fr;
    }
  }

  .socials {
    justify-self: end;
    align-self: end;
    padding: 0 1.5rem 2.5rem 0;

    @media only screen and (max-width: 600px) {
      grid-column: 1/-1;
      padding: 0;
      justify-self: center;
      padding-bottom: 1rem;
    }

    & a {
      margin: 0 0.25rem;

      & svg {
        font-size: 1.6rem;
        color: var(--color-green);
      }

      &:hover {
        filter: brightness(75%);
      }
    }
  }

  .copyright {
    border-top: 1px solid var(--color-green);
    padding: 0.75rem 0;
    display: grid;
    grid-template-columns: repeat(4, max-content);
    align-items: center;
    font-size: 14px;
    column-gap: 4rem;

    @media only screen and (max-width: 1000px) {
      column-gap: 1.5rem;
    }

    @media only screen and (max-width: 800px) {
      grid-template-columns: 1fr;
      row-gap: 0.5rem;
    }

    & span {
      @media only screen and (max-width: 800px) {
        margin-top: 0.5rem;
        grid-row: 4;
      }
      @media only screen and (max-width: 600px) {
        text-align: center;
      }
    }

    & a {
      color: var(--color-white);
      text-decoration: none;

      &:hover {
        text-decoration: underline;
      }
    }
  }
`;

export const FooterBlockWrapper = styled.div`
  display: grid;
  row-gap: 1rem;
  padding-bottom: 7rem;

  @media only screen and (max-width: 600px) {
    padding-bottom: 5rem;
  }

  @media only screen and (max-width: 400px) {
    padding-bottom: 3rem;
  }

  & h3 {
    font-size: 20px;
  }

  & ul {
    display: grid;
    row-gap: 1.25rem;

    & a {
      color: var(--color-white);
      text-decoration: none;
      font-weight: 300;

      &:hover {
        text-decoration: underline;
      }
    }
  }
`;
