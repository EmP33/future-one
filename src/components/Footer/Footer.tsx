import React from "react";
import { Link } from "react-router-dom";
import { Wrapper, FooterBlockWrapper } from "./Footer.styles";
// Icons
import {
  AiFillFacebook,
  AiOutlineTwitter,
  AiFillLinkedin,
  AiOutlineMedium,
  AiFillGithub,
} from "react-icons/ai";

interface IData {
  title: string;
  links: string[];
}

const data: IData[] = [
  {
    title: "Products",
    links: ["Tiramisu chocolate", "Pastry tart", "Pllipop brownie"],
  },
  {
    title: "Collaboration",
    links: ["Tiramisu chocolate", "Pastry tart", "Pllipop brownie"],
  },
];

const FooterBlock: React.FC<{ block: IData }> = ({
  block: { title, links },
}) => (
  <FooterBlockWrapper>
    <h3>{title}</h3>
    <ul>
      {links.map((link) => (
        <li key={link}>
          <Link to="/">{link}</Link>
        </li>
      ))}
    </ul>
  </FooterBlockWrapper>
);

const Footer = () => {
  return (
    <Wrapper>
      <div className="footer-wrapper">
        <div className="footer-content">
          {data.map((block) => (
            <FooterBlock key={block.title} block={block} />
          ))}
          <div className="socials">
            <a href="https://www.facebook.com/">
              <AiFillFacebook />
            </a>
            <a href="https://twitter.com/">
              <AiOutlineTwitter />
            </a>
            <a href="https://www.linkedin.com/">
              <AiFillLinkedin />
            </a>
            <a href="https://medium.com/">
              <AiOutlineMedium />
            </a>
            <a href="https://github.com/">
              <AiFillGithub />
            </a>
          </div>
        </div>
        <div className="copyright">
          <span>
            &copy; 2003 - 2022 Carrot cake sweet pie Sp. z o.o. All rights
            reserved.
          </span>
          <Link to="/">Legal</Link>
          <Link to="/">Privacy policy</Link>
          <Link to="/">Proud to be Open Source</Link>
        </div>
      </div>
    </Wrapper>
  );
};

export default Footer;
