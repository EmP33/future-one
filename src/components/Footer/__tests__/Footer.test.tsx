import Footer from "../Footer";
import ShallowRenderer from "react-test-renderer/shallow";

describe("<Footer />", () => {
  it("should render properly", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Footer />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
