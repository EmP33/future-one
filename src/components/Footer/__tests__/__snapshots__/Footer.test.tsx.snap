// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render properly 1`] = `
<styled.footer>
  <div
    className="footer-wrapper"
  >
    <div
      className="footer-content"
    >
      <FooterBlock
        block={
          Object {
            "links": Array [
              "Tiramisu chocolate",
              "Pastry tart",
              "Pllipop brownie",
            ],
            "title": "Products",
          }
        }
      />
      <FooterBlock
        block={
          Object {
            "links": Array [
              "Tiramisu chocolate",
              "Pastry tart",
              "Pllipop brownie",
            ],
            "title": "Collaboration",
          }
        }
      />
      <div
        className="socials"
      >
        <a
          aria-label="facebook"
          href="https://www.facebook.com/"
        >
          <AiFillFacebook />
        </a>
        <a
          aria-label="twitter"
          href="https://twitter.com/"
        >
          <AiOutlineTwitter />
        </a>
        <a
          aria-label="linkedin"
          href="https://www.linkedin.com/"
        >
          <AiFillLinkedin />
        </a>
        <a
          aria-label="medium"
          href="https://medium.com/"
        >
          <AiOutlineMedium />
        </a>
        <a
          aria-label="github"
          href="https://github.com/"
        >
          <AiFillGithub />
        </a>
      </div>
    </div>
    <div
      className="copyright"
    >
      <span>
        Â© 2003 - 2022 Carrot cake sweet pie Sp. z o.o. All rights reserved.
      </span>
      <Link
        to="/"
      >
        Legal
      </Link>
      <Link
        to="/"
      >
        Privacy policy
      </Link>
      <Link
        to="/"
      >
        Proud to be Open Source
      </Link>
    </div>
  </div>
</styled.footer>
`;
